radicle-node/src/service/message.rs

```diff
@@ -365,10 +376,22 @@ impl fmt::Debug for Message {
             Self::Announcement(Announcement { node, message, .. }) => {
                 write!(f, "Announcement({}, {:?})", node, message)
             }
+            Self::Ping { ponglen, zeroes } => write!(f, "Ping({ponglen}, {zeroes})"),
+            Self::Pong { zeroes } => write!(f, "Pong({zeroes})"),
         }
     }
 }
 
+// Should ZeroBytes be in wire/messsages instead?
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ZeroBytes(pub u16);
```

Comment:
I wouldn't expose the byte length publically, kind of defeats part of the purpose of this struct. I'd instead have a `len() -> usize` or `len() -> Size` method on it.

Reply:
Would this affect match statements on messages?

Reply:
Don't think so..

Reply:
There's a complication.  The match statement must be declarative (?) but making the field private requires using a public constructor.  Match statements can't contain functions (constructors).

Reply:
Why do you need to use a match statement though?

Reply:
What's being gained by making it private?  We'd be adding restrictions for no gains.

There may be confusion in the intent of the type.  It represents a length of zero bytes on the wire, not in memory.

Reply:
It is meant to be a type'd number, not similar to a struct.  Is using a single element tuple not the way to do this in rust?